src/huggingface_hub/hf_api.py

```diff
@@ -4517,27 +4591,39 @@ def upload_folder(
                 merge_pr=not create_pr,
                 verbose=multi_commits_verbose,
             )
-        else:
-            commit_info = self.create_commit(
-                repo_type=repo_type,
-                repo_id=repo_id,
-                operations=commit_operations,
-                commit_message=commit_message,
-                commit_description=commit_description,
-                token=token,
-                revision=revision,
-                create_pr=create_pr,
-                parent_commit=parent_commit,
-            )
-            pr_url = commit_info.pr_url
+            # Defining a CommitInfo object is not really relevant in this case
+            # Let's return early with pr_url only (as string).
+            return pr_url
+
+        commit_info = self.create_commit(
+            repo_type=repo_type,
+            repo_id=repo_id,
+            operations=commit_operations,
+            commit_message=commit_message,
+            commit_description=commit_description,
+            token=token,
+            revision=revision,
+            create_pr=create_pr,
+            parent_commit=parent_commit,
+        )
```

Comment:
why do you need this extra commit for the multi_commits case ?

Reply:
The function returns before this line is the multi-commit case: https://github.com/huggingface/huggingface_hub/blob/b37f20b91af4f5957ef23c4f42d4714a5fb28f40/src/huggingface_hub/hf_api.py#L4581-L4596

Reply:
Yes exactly. As @mariosasko said, the multi_commit is processed separately.